source/docs/advanced/features/external-logins.md

```diff
@@ -0,0 +1,314 @@
+
+### Configuring Redirect URIs
+
+You need to configure the allowed redirect URIs on your [facebook developer console](https://developers.facebook.com/apps/). To do so go you need to select your app and then go to `Facebook Login > Settings` section on the left menu. There you will find a field called `Valid OAuth Redirect URIs`. You need to fill this field with your stores URLs folowed by `/external-login/facebook/callback`. For example if you have 2 stores `https://example.com/en` and `https://example.com/fr`, you need to put `https://example.com/en/external-login/facebook/callback` and `https://example.com/fr/external-login/facebook/callback` in the `Valid OAuth Redirect URIs` field.
+
+### Working with the Facebook provider in development mode
+
+External login rely on redirections, and in devlopment your server is usally at `http://localhost` i.e. not reachable to Facebook. So to circumvent this in development mode you can use a proxying service such as [ngrok](https://ngrok.com/) to proxy requests to your local machine. A proxying service will pprovide a publically accessible URL to your local server. So to work with Facebook external logins on your local machine you need to set the `FRONT_COMMERCE_FACEBOOK_DEVELOPMENT_BASE_CALLBACK_URL` environment variable to the URL given to you by your proxying service. (p.s. use just the origin part of the url without any trailing slashes `/`).
+
+_Note: do not forget to [reconfigure your app](#Configuring-Redirect-URIs) with this new URL_
+
+After starting the proxying service and configuring your app and `FRONT_COMMERCE_FACEBOOK_DEVELOPMENT_BASE_CALLBACK_URL` accordingly, the "Login with Facebook" button should work properly.
+
+## Configuring the Google provider
+
+The `GoogleProvider` needs the `clientId` and the `clientSecret` to work. You can get these from your "OAuth2.0 Client" of your app on the [google developer console](https://console.developers.google.com/apis/credentials). P.S. you need to create one if you have not already.
+
+To configure the `clientId` and the `clientSecret` you need to define the folowing environment variables:
+* `FRONT_COMMERCE_GOOGLE_CLIENT_ID` for the `clientId`
+* `FRONT_COMMERCE_GOOGLE_CLIENT_SECRET` for the `clientSecret`
+
+### Configuring Redirect URIs
+
+You need to configure the allowed redirect URIs under your "OAuth2.0 Client" of your app on the [google developer console](https://console.developers.google.com/apis/credentials). Fill the `Authorized redirect URIs` field with your stores URLs folowed by `/external-login/google/callback`. For example if you have 2 stores `https://example.com/en` and `https://example.com/fr`, you need to put `https://example.com/en/external-login/google/callback` and `https://example.com/fr/external-login/google/callback` in the `Authorized redirect URIs` field. Also set the `Authorized JavaScript origins` on the same page (in our example above that would be just `https://example.com`).
```

Comment:
```suggestion
You need to configure the allowed redirect URIs under your "OAuth2.0 Client" of your app on the [Google developer console](https://console.developers.google.com/apis/credentials). Fill the `Authorized redirect URIs` field with your stores URLs followed by `/external-login/google/callback`. For example if you have 2 stores `https://example.com/en` and `https://example.com/fr`, you need to put `https://example.com/en/external-login/google/callback` and `https://example.com/fr/external-login/google/callback` in the `Authorized redirect URIs` field. Also set the `Authorized JavaScript origins` on the same page (in our example above that would be just `https://example.com`).
```

(you should consider installing a spell checker to avoid similar mistakes)